### PR TITLE
Implemented PDF exporting functional for users' orders

### PR DESCRIPTION
**Pull request by Andrii-Danylenko**
## GIT

* [Main GIT ticket](https://github.com/ita-social-projects/GreenCity/issues/8202)

I have implemented the ability to export order data to a PDF file with support for Ukrainian and English locales.

image

image

## Summary by CodeRabbit

- **New Features**
	- Added a new capability for users to export detailed order summaries as PDF files.
	- PDF exports now support multiple languages, delivering localized content in English and Ukrainian.
	- Enhanced PDF formatting provides clear, structured order details with improved layout and error notifications during the export process.

### CODE REVIEW

**Review by maks741 — APPROVED** [review]

File: core/src/main/java/greencity/configuration/SecurityConfig.java [hunk 1]
```diff
@@ -318,6 +319,8 @@ public SecurityFilterChain securityFilterChain(HttpSecurity http) throws Excepti
                 .hasAnyRole(USER, ADMIN)
                 .requestMatchers(HttpMethod.HEAD,
                     UBS_LINK + "/userProfile/**")
+                .hasAnyRole(USER, ADMIN)
+                .requestMatchers(HttpMethod.GET, UBS_EXPORT)
```
Comment (VitaliiChernenko) [inline core/src/main/java/greencity/configuration/SecurityConfig.java]: It is not necessary to add new matches. In this SecurityConfig already present matches for your purposes.
For example in lines 297 -300(in this moment) present for GET requests for roles USER and ADMIN. You can add your endpoint URL there.

File: core/src/main/java/greencity/controller/OrderPdfExporterController.java [hunk 2]
```diff
@@ -0,0 +1,53 @@
...
+import org.springframework.core.io.Resource;
+import org.springframework.web.bind.annotation.GetMapping;
+import org.springframework.web.bind.annotation.RequestParam;
+import java.util.Locale;
+import java.util.UUID;
+
+@RestController
+@RequestMapping("ubs/order/pdf/export")
+@AllArgsConstructor
+@Slf4j
```
Comment (VitaliiChernenko) [inline core/src/main/java/greencity/controller/OrderPdfExporterController.java]: Why do you need it?
Where do You use it?

**Review by softservedata — APPROVED** [review]

File: core/src/main/java/greencity/controller/OrderPdfExporterController.java [hunk 3]
```diff
@@ -0,0 +1,53 @@
...
+public class OrderPdfExporterController {
+    private final PdfExporterService pdfExporterService;
+
+    @Operation(summary = "Returns PDF file with requested order.")
+    @ApiResponses(value = {
+        @ApiResponse(responseCode = "200", description = HttpStatuses.OK),
+        @ApiResponse(responseCode = "400", description = HttpStatuses.BAD_REQUEST, content = @Content),
+        @ApiResponse(responseCode = "401", description = HttpStatuses.UNAUTHORIZED, content = @Content),
+        @ApiResponse(responseCode = "403", description = HttpStatuses.FORBIDDEN, content = @Content),
```
Comment (VitaliiChernenko) [inline core/src/main/java/greencity/controller/OrderPdfExporterController.java]: Can this controller return 403?

File: core/src/main/java/greencity/controller/OrderPdfExporterController.java [hunk 4]
```diff
@@ -0,0 +1,53 @@
...
+    })
+    @GetMapping(produces = MediaType.APPLICATION_PDF_VALUE)
+    public ResponseEntity<Resource> exportOrderPdf(@Parameter(hidden = true) @CurrentUserUuid String userUuid,
+        @Parameter @RequestParam("id") Long orderId,
+        @Parameter @RequestParam(name = "lang") @ValidLanguage Locale locale) {
+        Resource resource = pdfExporterService.exportById(orderId, locale, userUuid);
+        HttpHeaders headers = new HttpHeaders();
+        headers.setContentType(MediaType.parseMediaType("application/pdf"));
+        headers.add(HttpHeaders.CONTENT_DISPOSITION, "attachment; filename=order_details_%d_%s_%s.pdf"
+            .formatted(orderId, UUID.randomUUID(), locale.getLanguage()));
```
Comment (VitaliiChernenko) [inline core/src/main/java/greencity/controller/OrderPdfExporterController.java]: Why do we need in the file name `UID.randomUUID`?

File: core/src/test/java/greencity/controller/OrderPdfExporterControllerTest.java [hunk 5]
```diff
@@ -0,0 +1,68 @@
...
+import org.mockito.junit.jupiter.MockitoExtension;
+import org.springframework.boot.web.servlet.error.DefaultErrorAttributes;
+import org.springframework.boot.web.servlet.error.ErrorAttributes;
+import org.springframework.context.annotation.Import;
+import org.springframework.data.web.PageableHandlerMethodArgumentResolver;
+import org.springframework.http.MediaType;
+import org.springframework.test.web.servlet.MockMvc;
+import org.springframework.test.web.servlet.setup.MockMvcBuilders;
+import static org.mockito.Mockito.times;
+import static org.mockito.Mockito.verify;
```
Comment (VitaliiChernenko) [inline core/src/test/java/greencity/controller/OrderPdfExporterControllerTest.java:18]: It is not necessary but in my opinion, static imports should be separated from usual.

File: core/src/test/java/greencity/controller/OrderPdfExporterControllerTest.java [hunk 6]
```diff
@@ -0,0 +1,68 @@
...
+import java.security.Principal;
+import java.util.Locale;
+import java.util.UUID;
+import static greencity.ModelUtils.getUuid;
+import static org.springframework.test.web.servlet.result.MockMvcResultMatchers.header;
+import static org.springframework.test.web.servlet.result.MockMvcResultMatchers.content;
+import static org.springframework.test.web.servlet.result.MockMvcResultMatchers.status;
+
+@ExtendWith(MockitoExtension.class)
+@Import(SecurityConfig.class)
```
Comment (VitaliiChernenko) [inline core/src/test/java/greencity/controller/OrderPdfExporterControllerTest.java]: Why do you need `@Import(SecurityConfig.class)`?
It looks like it does nothing here.

File: core/src/test/java/greencity/controller/OrderPdfExporterControllerTest.java [hunk 7]
```diff
@@ -0,0 +1,68 @@
...
+    @BeforeEach
+    void setUp() {
+        mockMvc = MockMvcBuilders.standaloneSetup(controller).setCustomArgumentResolvers(
+            new PageableHandlerMethodArgumentResolver())
+            .setControllerAdvice(new CustomExceptionHandler(errorAttributes))
+            .build();
+    }
+
+    @Test
+    void exportTest() throws Exception {
```
Comment (VitaliiChernenko) [inline core/src/test/java/greencity/controller/OrderPdfExporterControllerTest.java]: Please rename the method according to our test method convention.
https://github.com/ita-social-projects/GreenCity/wiki/Naming-Conventions

For example **exportOrderAsPdfWithValidParamsTest** or similar.

File: service-api/src/main/java/greencity/exceptions/exporting/pdf/PdfFileExportingException.java [hunk 8]
```diff
@@ -0,0 +1,14 @@
+package greencity.exceptions.exporting.pdf;
+
+public class PdfFileExportingException extends RuntimeException {
+    public PdfFileExportingException() {
+    }
+
+    public PdfFileExportingException(String message) {
+        super(message);
+    }
```
Comment (VitaliiChernenko) [inline service-api/src/main/java/greencity/exceptions/exporting/pdf/PdfFileExportingException.java]: Do You need constructors without params and with one param?
It looks like you never use them.

Think about replacing all these constructors with the `@StandardException` annotation on top of the exception class. As for me it will be more attractive.

File: service-api/src/main/java/greencity/service/ubs/PdfExporterService.java [hunk 9]
```diff
@@ -0,0 +1,8 @@
+package greencity.service.ubs;
+
+import org.springframework.core.io.Resource;
+import java.util.Locale;
+
+public interface PdfExporterService {
+    Resource exportById(long orderId, Locale locale, String userUuid);
```
Comment (VitaliiChernenko) [inline service-api/src/main/java/greencity/service/ubs/PdfExporterService.java]: It would be great if You will add a java doc for this method.

Comment (VitaliiChernenko) [inline service-api/src/main/java/greencity/service/ubs/PdfExporterService.java]: As for me, the method name is not quite good.
It is not really obvious what this PDF (order or something other) is.

Think about renaming it more clearly. For example _generateOrderPdfById_ or _exportOrderPdfById_ or something.

File: service/pom.xml [hunk 10]
```diff
@@ -163,5 +163,10 @@
             <artifactId>org.eclipse.jgit</artifactId>
             <version>${org.eclipse.jgit.version}</version>
         </dependency>
+        <dependency>
+            <groupId>com.github.librepdf</groupId>
+            <artifactId>openpdf</artifactId>
+            <version>2.0.3</version>
```
Comment (VitaliiChernenko) [inline service/pom.xml]: Please move the lib version in the properties section.

File: service/src/main/java/greencity/constant/AppConstant.java [hunk 11]
```diff
@@ -36,4 +37,7 @@ public class AppConstant {
 
     public static final String UNKNOWN_ENG = "Unknown";
     public static final String UNKNOWN_UA = "Невідомо";
+
```
Comment (VitaliiChernenko) [inline service/src/main/java/greencity/constant/AppConstant.java]: Why did You leave an empty line here?
Is it necessary here?

File: service/src/main/java/greencity/constant/pdf/PdfAddressConstants.java [hunk 12]
```diff
@@ -0,0 +1,26 @@
...
+import lombok.Getter;
+import lombok.RequiredArgsConstructor;
+import java.util.Locale;
+import java.util.Objects;
+
+@Getter
+@RequiredArgsConstructor
+public enum PdfAddressConstants {
+    HOUSE_NUMBER("буд.", "b."),
+    HOUSE_CORPUS_NUMBER("корпус ", "b."),
```
Comment (VitaliiChernenko) [inline service/src/main/java/greencity/constant/pdf/PdfAddressConstants.java]: Is it ok that for the _HOUSE_NUMBER_ and _HOUSE_CORPUS_NUMBER_ we have the same values **b**?

File: service/src/main/java/greencity/constant/pdf/PdfAddressConstants.java [hunk 13]
```diff
@@ -0,0 +1,26 @@
...
+    DISTRICT("район", "d.");
+
+    private final String nameUa;
+    private final String nameEn;
+
+    public static String getByLocale(PdfAddressConstants headerName, Locale locale) {
+        if (Objects.equals(AppConstant.LOCALE_ENG_NAME, locale.getLanguage())) {
+            return headerName.nameEn;
+        }
+        return headerName.nameUa;
```
Comment (ChernenkoVitaliy) [inline service/src/main/java/greencity/constant/pdf/PdfAddressConstants.java:24]: What if will be passed not **ua** and not **en** locale? PDF will be as _ua_ lang as default?

I'm not sure, but maybe it makes sense to throw an exception if we don't have such language.

File: service/src/main/java/greencity/service/ubs/pdf/exporter/OrdersDataPdfFileExporterImpl.java [hunk 14]
```diff
@@ -0,0 +1,245 @@
...
+            document.add(fillOrderInfoTable(objectToWrite, locale, tableOrderDetails, DEFAULT_PARAGRAPH_FONT_SIZE));
+            addLineSeparator(document);
+            PdfPTable table = createTable(PdfOrderContentDetailsHeaders.getAllByLocale(locale),
+                DEFAULT_TABLE_HEADER_FONT_SIZE,
+                ORDER_CONTENT_TABLE_COLUMN_WIDTH);
+            document.add(fillOrdersTable(objectToWrite.getBags(), locale, table, DEFAULT_PARAGRAPH_FONT_SIZE));
+            addNewLine(document);
+            addLineSeparator(document);
+            addFooter(objectToWrite, locale, document);
+            document.close();
```
Comment (VitaliiChernenko) [inline service/src/main/java/greencity/service/ubs/pdf/exporter/OrdersDataPdfFileExporterImpl.java:81]: If an exception is thrown before this line `document.close();` will the resource be closed?
In my opinion `document.close();` should be moved to the **finally** block.

File: service/src/main/java/greencity/service/ubs/pdf/exporter/OrdersDataPdfFileExporterImpl.java [hunk 15]
```diff
@@ -0,0 +1,245 @@
...
+            return byteArrayOutputStream.toByteArray();
+        } catch (IOException e) {
+            throw new PdfFileExportingException(CANNOT_EXPORT_DATA_TO_PDF, e);
```
Comment (VitaliiChernenko) [inline service/src/main/java/greencity/service/ubs/pdf/exporter/OrdersDataPdfFileExporterImpl.java:84]: Please add unit test for this case.

File: service/src/main/java/greencity/service/ubs/pdf/exporter/PdfExporterServiceImpl.java [hunk 16]
```diff
@@ -0,0 +1,24 @@
...
+import greencity.service.ubs.file.export.FileExporter;
+import lombok.RequiredArgsConstructor;
+import org.springframework.core.io.ByteArrayResource;
+import org.springframework.core.io.Resource;
+import org.springframework.stereotype.Service;
+import java.util.Locale;
+
+@Service
+@RequiredArgsConstructor
+public class PdfExporterServiceImpl implements PdfExporterService {
```
Comment (VitaliiChernenko) [inline service/src/main/java/greencity/service/ubs/pdf/exporter/PdfExporterServiceImpl.java]: Please add unit tests for this service.

File: service/src/test/java/greencity/ModelUtils.java [hunk 17]
```diff
@@ -54,45 +55,18 @@
...
-import greencity.dto.order.OtherPackages;
-import greencity.dto.order.ReadAddressByOrderDto;
-import greencity.dto.order.RequestToChangeOrdersDataDto;
-import greencity.dto.order.SenderLocation;
-import greencity.dto.order.UpdateAllOrderPageDto;
-import greencity.dto.order.UpdateOrderDetailDto;
-import greencity.dto.order.UpdateOrderPageAdminDto;
+import greencity.dto.notification.NotificationTemplateDto;
+import greencity.dto.notification.SenderInfoDto;
+import greencity.dto.order.*;
```
Comment (VitaliiChernenko) [inline service/src/test/java/greencity/ModelUtils.java:69]: You are not supposed to use wildcard imports.
Please replace them with simple ones.

File: service/src/test/java/greencity/ModelUtils.java [hunk 18]
```diff
@@ -5870,4 +5844,74 @@ public static OrderAddress getOrderAddress1() {
             .addressComment("Test comment for address №1")
             .build();
     }
+
+    public static BagForUserDto getBagForUserDto() {
```
Comment (VitaliiChernenko) [inline service/src/test/java/greencity/ModelUtils.java:5848]: In the current class already presents a method for creating `BagForUserDto` with the name `createBagForUserDto`.
Maybe You can use that instead of creating a new one?

File: service/src/test/java/greencity/service/ubs/PdfExporterServiceTest.java [hunk 19]
```diff
@@ -0,0 +1,45 @@
...
+import static org.junit.jupiter.api.Assertions.assertArrayEquals;
+import static org.mockito.ArgumentMatchers.anyLong;
+import static org.mockito.ArgumentMatchers.anyString;
+import static org.mockito.ArgumentMatchers.any;
+import static org.mockito.Mockito.when;
+import static org.mockito.Mockito.verify;
+import static org.mockito.Mockito.times;
+
+@ExtendWith(MockitoExtension.class)
+class PdfExporterServiceTest {
```
Comment (VitaliiChernenko) [inline service/src/test/java/greencity/service/ubs/PdfExporterServiceTest.java]: Please, move this class into the right package(_service.ubs.exporter.pdf_) and rename it as `PdfExporterServiceImplTest`.

File: service/src/main/java/greencity/service/ubs/pdf/exporter/PdfExporterServiceImpl.java (hunk 16 above)
Comment (VitaliiChernenko) [reply service/src/main/java/greencity/service/ubs/pdf/exporter/PdfExporterServiceImpl.java]: My mistake. I've found tests for this class but they are in the wrong place and with a bit different name as supposed to be.

File: service/src/test/java/greencity/service/ubs/PdfExporterServiceTest.java [hunk 20]
```diff
@@ -0,0 +1,45 @@
...
+    @Mock
+    private FileExporter<OrdersDataForUserDto> fileExporterService;
+    @Mock
+    private UBSClientService ubsClientService;
+    @InjectMocks
+    private PdfExporterServiceImpl pdfExporterService;
+
+    @Test
+    void testExportById() {
```
Comment (VitaliiChernenko) [inline service/src/test/java/greencity/service/ubs/PdfExporterServiceTest.java]: Please rename test using project's naming convention
https://github.com/ita-social-projects/GreenCity/wiki/Naming-Conventions

File: service/src/test/java/greencity/service/ubs/PdfExporterServiceTest.java [hunk 21]
```diff
@@ -0,0 +1,45 @@
...
+import greencity.service.ubs.file.export.FileExporter;
+import greencity.service.ubs.pdf.exporter.PdfExporterServiceImpl;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.extension.ExtendWith;
+import org.mockito.InjectMocks;
+import org.mockito.Mock;
+import org.mockito.junit.jupiter.MockitoExtension;
+import org.springframework.core.io.ByteArrayResource;
+import org.springframework.core.io.Resource;
+
```
Comment (VitaliiChernenko) [inline service/src/test/java/greencity/service/ubs/PdfExporterServiceTest.java]: An empty row can be removed.

File: service/src/test/java/greencity/service/ubs/PdfExporterServiceTest.java [hunk 22]
```diff
@@ -0,0 +1,45 @@
...
+    void testExportById() {
+        OrdersDataForUserDto orderToWrite = ModelUtils.getOrdersDataForUserDto();
+        byte[] pdfBytes = {1, 2, 3};
+        when(ubsClientService.getOrderForUser(anyString(), anyLong())).thenReturn(orderToWrite);
+        when(fileExporterService.export(any(), any())).thenReturn(pdfBytes);
+        Resource resource = pdfExporterService.exportById(1L, Locale.ENGLISH, "user-uuid");
+        assertArrayEquals(pdfBytes, ((ByteArrayResource) resource).getByteArray());
+        verify(ubsClientService, times(1)).getOrderForUser("user-uuid", 1L);
+        verify(fileExporterService, times(1)).export(orderToWrite, Locale.ENGLISH);
+    }
```
Comment (VitaliiChernenko) [inline service/src/test/java/greencity/service/ubs/PdfExporterServiceTest.java]: Please add tests for negative scenarios.
For example: if `orderId` is not valid(negative) or `userUuid` doesn't exist.

File: service/src/test/java/greencity/service/ubs/exporter/pdf/OrdersDataPdfFileExporterImplTest.java [hunk 23]
```diff
@@ -0,0 +1,76 @@
...
+class OrdersDataPdfFileExporterImplTest {
+    private OrdersDataPdfFileExporterImpl pdfFileExporter;
+
+    @BeforeEach
+    void setUp() {
+        pdfFileExporter = new OrdersDataPdfFileExporterImpl();
+    }
+
+    @Test
+    void exportValidEnPdf() throws IOException {
```
Comment (VitaliiChernenko) [inline service/src/test/java/greencity/service/ubs/exporter/pdf/OrdersDataPdfFileExporterImplTest.java:33]: Please rename tests using project's naming convention
https://github.com/ita-social-projects/GreenCity/wiki/Naming-Conventions

**Review by RomaMocherniuk — APPROVED** [review]

**Comment by ChernenkoVitaliy** [comment]
Why do we have the en version short names for these fields but in the ua full?
short_form

File: core/src/main/java/greencity/exception/handler/CustomExceptionHandler.java [hunk 24]
```diff
@@ -125,6 +125,7 @@ public final ResponseEntity<Object> handleUnprocessableEntityException(Unprocess
     @ExceptionHandler({NotFoundException.class})
     public final ResponseEntity<Object> handleNotFoundException(NotFoundException exception, WebRequest request) {
         ExceptionResponse exceptionResponse = new ExceptionResponse(getErrorAttributes(request));
+        System.out.println("AN ERROR OCCURRED HELLO");
```
Comment (VitaliiChernenko) [inline core/src/main/java/greencity/exception/handler/CustomExceptionHandler.java]: Please remove `System.out.println` from code.

**Comment by Andrii-Danylenko** [comment]
> ## [Quality Gate Passed](https://sonarcloud.io/dashboard?id=ita-social-projects-green-ubs&pullRequest=1619) **Quality Gate passed**
> Issues  [1 New issue](https://sonarcloud.io/project/issues?id=ita-social-projects-green-ubs&pullRequest=1619&issueStatuses=OPEN,CONFIRMED&sinceLeakPeriod=true)  [0 Accepted issues](https://sonarcloud.io/project/issues?id=ita-social-projects-green-ubs&pullRequest=1619&issueStatuses=ACCEPTED)
> 
> Measures  [0 Security Hotspots](https://sonarcloud.io/project/security_hotspots?id=ita-social-projects-green-ubs&pullRequest=1619&issueStatuses=OPEN,CONFIRMED&sinceLeakPeriod=true)  [97.6% Coverage on New Code](https://sonarcloud.io/component_measures?id=ita-social-projects-green-ubs&pullRequest=1619&metric=new_coverage&view=list)  [0.0% Duplication on New Code](https://sonarcloud.io/component_measures?id=ita-social-projects-green-ubs&pullRequest=1619&metric=new_duplicated_lines_density&view=list)
> 
> [See analysis details on SonarQube Cloud](https://sonarcloud.io/dashboard?id=ita-social-projects-green-ubs&pullRequest=1619)

SonarCloud shows an error related to document.close(). This is not a bug or problem, since this method is necessary to generate the file. Without it, documentListeners will not be able to create the file

File: service/src/main/java/greencity/constant/AppConstant.java [hunk 25]
```diff
@@ -27,13 +28,13 @@ public class AppConstant {
...
     public static final String NOTIFICATOR_START_IS_FAILED_LOG_MESSAGE =
         "Failed to start scheduled notificator with type {} because cron is incorrect or template is inactive.";
     public static final String NOTIFICATOR_SUCCESSFULLY_START_LOG_MESSAGE =
         "Scheduled notificator for {} notification template by {} cron";
     public static final String NOTIFICATOR_RESTART_LOG_MESSAGE = "Restarting scheduled notificator {}";
-
     public static final String UNKNOWN_ENG = "Unknown";
     public static final String UNKNOWN_UA = "Невідомо";
+    public static final String LOCALE_UK_NAME = "ua";
+    public static final String LOCALE_ENG_NAME = "en";
```
Comment (oleksandra-bulhakova) [inline service/src/main/java/greencity/constant/AppConstant.java]: Please. rename field LOCALE_ENG_NAME to LOCALE_EN_NAME

File: service/src/main/java/greencity/constant/pdf/PdfAddressConstants.java [hunk 26]
```diff
@@ -0,0 +1,26 @@
...
+
+@Getter
+@RequiredArgsConstructor
+public enum PdfAddressConstants {
+    HOUSE_NUMBER("буд.", "b."),
+    HOUSE_CORPUS_NUMBER("корпус", "housing"),
+    ENTRANCE_NUMBER("під'їзд", "entrance"),
+    DISTRICT("район", "district");
+
+    private final String nameUa;
```
Comment (oleksandra-bulhakova) [inline service/src/main/java/greencity/constant/pdf/PdfAddressConstants.java]: Please, rename  private final String nameUa to nameUk

File: service/src/main/java/greencity/constant/pdf/PdfFileHeaders.java [hunk 27]
```diff
@@ -0,0 +1,25 @@
...
+
+@Getter
+@RequiredArgsConstructor
+public enum PdfFileHeaders {
+    ORDER_DETAILS("Деталі замовлення", "Order details"),
+    SENDER_INFO("Відправник", "Sender"),
+    ORDER_COMMENT("Коментар до замовлення", "Comment to the order"),
+    ADDRESS_INFO("Адреса вивезення відходів", "The address of export of the ordered services");
+
+    private final String nameUa;
```
Comment (oleksandra-bulhakova) [inline service/src/main/java/greencity/constant/pdf/PdfFileHeaders.java]: Please, rename private final String nameUa to nameUk

File: service/src/main/java/greencity/constant/pdf/PdfOrderContentDetailsHeaders.java [hunk 28]
```diff
@@ -0,0 +1,49 @@
...
+    QUANTITY("Кількість пакетів, "
+        + PdfUnitsOfMeasurement.getByLocale(PdfUnitsOfMeasurement.UNITS, Locale.of(AppConstant.LOCALE_UK_NAME)),
+             "Quantity of bags, " + PdfUnitsOfMeasurement.getByLocale(PdfUnitsOfMeasurement.UNITS,
+                 Locale.of(AppConstant.LOCALE_ENG_NAME))),
+    SUM("Сума, "
+        + PdfUnitsOfMeasurement.getByLocale(PdfUnitsOfMeasurement.CURRENCY, Locale.of(AppConstant.LOCALE_UK_NAME)),
+        "Sum, " + PdfUnitsOfMeasurement.getByLocale(PdfUnitsOfMeasurement.CURRENCY,
+            Locale.of(AppConstant.LOCALE_ENG_NAME)));
+
+    private final String nameUa;
```
Comment (oleksandra-bulhakova) [inline service/src/main/java/greencity/constant/pdf/PdfOrderContentDetailsHeaders.java]: Please, rename nameUa to nameUk

File: service/src/main/java/greencity/constant/pdf/PdfOrderDetailsHeaders.java [hunk 29]
```diff
@@ -0,0 +1,44 @@
...
+    PAYMENT_AMOUNT("Сума замовлення, "
+        + PdfUnitsOfMeasurement.getByLocale(PdfUnitsOfMeasurement.CURRENCY, Locale.of(AppConstant.LOCALE_UK_NAME)),
+                   "Payment amount, " + PdfUnitsOfMeasurement.getByLocale(PdfUnitsOfMeasurement.CURRENCY,
+                       Locale.of(AppConstant.LOCALE_ENG_NAME))),
+    AMOUNT_DUE("Сума до оплати, "
+        + PdfUnitsOfMeasurement.getByLocale(PdfUnitsOfMeasurement.CURRENCY, Locale.of(AppConstant.LOCALE_UK_NAME)),
+               "Amount-due, " + PdfUnitsOfMeasurement.getByLocale(PdfUnitsOfMeasurement.CURRENCY,
+                   Locale.of(AppConstant.LOCALE_ENG_NAME)));
+
+    private final String nameUa;
```
Comment (oleksandra-bulhakova) [inline service/src/main/java/greencity/constant/pdf/PdfOrderDetailsHeaders.java]: Please, rename nameUa to nameUk

File: service/src/main/java/greencity/constant/pdf/PdfUnitsOfMeasurement.java [hunk 30]
```diff
@@ -0,0 +1,25 @@
...
+import java.util.Objects;
+
+@Getter
+@RequiredArgsConstructor
+public enum PdfUnitsOfMeasurement {
+    VOLUME("л", " l"),
+    UNITS("шт.", " pc."),
+    CURRENCY("грн", "UAH");
+
+    private final String nameUa;
```
Comment (oleksandra-bulhakova) [inline service/src/main/java/greencity/constant/pdf/PdfUnitsOfMeasurement.java]: Please, rename nameUa to nameUk

**Review by oleksandra-bulhakova — APPROVED** [review]
